NextGenConnect/JSON to HL7v2.xml

```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
-*/</script>
+
+// PID-22 Ethnic Group -- ??? Try to grab value ?
+
+//PID-24 Multiple Birth Indicator
+var birthBoolean = patient[&apos;multipleBirthBoolean&apos;];
+
+if(birthBoolean === false){
+	tmp[&apos;PID&apos;][&apos;PID.24&apos;][&apos;PID.24.1&apos;] = &apos;N&apos;;
+	}
+else{
+	tmp[&apos;PID&apos;][&apos;PID.24&apos;][&apos;PID.24.1&apos;] = &apos;Y&apos;;
+}</script>
         </com.mirth.connect.plugins.javascriptstep.JavaScriptStep>
         <com.mirth.connect.plugins.javascriptstep.JavaScriptStep version="3.10.0">
-          <name>HTTP request</name>
+          <name>PD1 Segment (RE)</name>
           <sequenceNumber>2</sequenceNumber>
           <enabled>true</enabled>
-          <script>// Create url object
+          <script>// WIR message may have 1 PD1 Segment
+
+//PD1-11 - Publicity Code
+
+
+//PD1-12 - Protection Indicator
+
+
+//PD1-13 Effective date for PD1-12
+
+
+//PD1-16 Immunization Registry Status
+
+
+//PD1-17 Effective date for PD1-16
+
+
+//PD1-18 Effective date for PD1-11</script>
+        </com.mirth.connect.plugins.javascriptstep.JavaScriptStep>
+        <com.mirth.connect.plugins.javascriptstep.JavaScriptStep version="3.10.0">
+          <name>NK1 Segment (RE)</name>
+          <sequenceNumber>3</sequenceNumber>
+          <enabled>true</enabled>
+          <script>/*
+// WIR message may have 0+ NK1 Segments
+// Currently deciding to omit this segment
+
+
+// NK1-1 Set ID - Sets the ID for the first related party and so on 1..2..3 for multiple NK1 segments
+// If only listing one contact this always remains 1
+
+tmp[&apos;NK1&apos;][&apos;NK1.1&apos;] = 1;
+
+//NK1-2 Name of responsible person
+
+tmp[&apos;NK1&apos;][&apos;NK1.2&apos;][&apos;NK1.2.1&apos;] = patient[&apos;contact&apos;][1][&apos;family&apos;];
+tmp[&apos;NK1&apos;][&apos;NK1.2&apos;][&apos;NK1.2.2&apos;] = patient[&apos;contact&apos;][1][&apos;given&apos;][0]; // Stored in array??
+tmp[&apos;NK1&apos;][&apos;NK1.2&apos;][&apos;NK1.2.7&apos;] = &quot;L&quot;;  // Legal Name
+
+//NK1-3 Relationship -- Trick one because of 0063 in WIR and 0131 aren&apos;t compatible in FHIR... use something else in FHIR?
+
+
+tmp[&apos;NK1&apos;][&apos;NK1.3&apos;][&apos;NK1.3.1&apos;] = patient[&apos;contact&apos;][0][0][1];
+
+// Need a bunch of if statements accordint to page 44 in WIR implementation guide if relationship == &quot;brother&quot; this = BRO
+// if relationship == cargeiver, this = CGV, and so forth
+
+
+tmp[&apos;NK1&apos;][&apos;NK1.3&apos;][&apos;NK1.3.2&apos;] = &apos;Mother&apos;; // Changes depending on code ^^^^
+
+tmp[&apos;NK1&apos;][&apos;NK1.3&apos;][&apos;NK1.3.3&apos;] = &apos;HL70063&apos;; // This is the default table... maybe find one compatible with FHIR 0131
+
+//NK1-4 Address of responsible person
+
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.2&apos;] = patient[&apos;contact&apos;][3][&apos;line&apos;][0];
```

Comment:
Same as PID: we need all the lines, else it might not be right.

Reply:
I just left this as we are not worrying about NK1

Reply:
Might as well fix it while we're here, right? You can just use the same solution as the one for PID, so it won't be much extra effort.